Slider/Assets/Scripts/Map/Mountain/Meltable.cs

```diff
@@ -4,47 +4,146 @@
+    private STile sTile;
+
 
-    // Update is called once per frame
-    void Update()
+    private void OnEnable() {
+        sTile = GetComponentInParent<MountainSTile>();
+        SGridAnimator.OnSTileMoveEndEarly += CheckFreezeOnMoveEnd; //C: Has to be early + delay or else tile in args is null
+    }
+
+    private void OnDisable() {
+        SGridAnimator.OnSTileMoveEndEarly -= CheckFreezeOnMoveEnd;
+    }
+
+    public void CheckFreezeOnMoveEnd(object sender, SGridAnimator.OnTileMoveArgs e)
     {
-        if(this.transform.position.y > 62 && !isFrozen && numLavaSources <= 0)
-            Freeze();
+        if(e.stile == null)
+            return;
+        if(e.stile == sTile)
+            StartCoroutine(WaitThenCheckFreeze());
     }
 
-    public void Melt()
+    //C: timing/collider jank
+    IEnumerator WaitThenCheckFreeze() 
     {
-        numLavaSources++;
-        if(isFrozen && numLavaSources > 0) //C: the second check is pointless but maybe wacky things could happen
+        yield return new WaitForSeconds(0.03f);
+        if(CheckFreeze())
+                Freeze();
+    }
+
+    public bool CheckFreeze()
+    {
+        return (!refreezeOnTop || sTile.y > 1) && (!anchorBroken || refreezeFromBroken) && numLavaSources <= 0;
+    }
+
+    public void Break(bool fromLoad = false) {
+        if(fromLoad || ((isFrozen || canBreakWhenNotFrozen) && canBreakWithAnchor && !anchorBroken))
+        {
+            anchorBroken = true;
+            numTimesBroken++;
+            isFrozen = false;
+           // animator.SetBool("Broken", true);
+            if(spriteRenderer)
+                spriteRenderer.sprite = anchorBrokenSprite;
+            onBreak.Invoke();
+        }
+    }
+
+    public void Melt(bool fromLoad = false)
+    {
+        if(fromLoad || (isFrozen && numLavaSources > 0)) 
         {
-            if (spriteRenderer)
-                spriteRenderer.sprite = meltedSprite;
             isFrozen = false;
+            if(spriteRenderer)
+                spriteRenderer.sprite = meltedSprite;
             onMelt.Invoke();
         }
     }
 
+    public void Freeze(bool fromLoad = false)
+    {
+        if(fromLoad || !isFrozen)
+        {
+            isFrozen = true;
+            if(spriteRenderer)
+                spriteRenderer.sprite = frozenSprite;
+            onFreeze.Invoke();
+        }
+    }
+
+    public void Fix()
+    {
+        if(anchorBroken) 
+        {
+            anchorBroken = false;
+           // animator.SetBool("Broken", false);
+            if(fixBackToFrozen)
+                Freeze();
+            else if(spriteRenderer)
+                spriteRenderer.sprite = meltedSprite;
+            onFix.Invoke();
+        }
+    }
+
     public void RemoveLava()
     {
         numLavaSources--;
     }
 
+    public void AddLava()
+    {
+        numLavaSources++;
+        Melt();
+    }
+
+    public void Save()
+    {
+        SaveSystem.Current.SetBool(gameObject.name + " frozen", isFrozen);
```

Comment:
nit: i think we are trying to have the convention for saved objects names be `camelCaseIfPossible`